### PR TITLE
Make a hook to mark forward and backward computation for CUDA profiler

### LINKED ISSUE #2209: Make a hook for CUDA profiler
`cupy.cuda.nvtx.RangePush` is very useful for performance tuning. I need a hook to call `RangePush` to check all forward and backward calculation of functions.

### PR DESCRIPTION
**Pull request by unnonouno**
fix #2209

### CODE REVIEW

File: chainer/function_hooks/cuda_profile.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from chainer import cuda
+from chainer import function
+
+if cuda.available:
+    from cupy.cuda import nvtx
```
Comment (okuta) [inline chainer/function_hooks/cuda_profile.py]: Sometime, Windows environment don't have `nvtx` module.
Please use try catch.

Comment (unnonouno) [inline chainer/function_hooks/cuda_profile.py]: What happens in such case? `cupy.cuda.nvtx.dll` is not compiled?

Comment (okuta) [inline chainer/function_hooks/cuda_profile.py]: Chainer do not create `cupy.cuda.nvtx` module.

Comment (unnonouno) [inline chainer/function_hooks/cuda_profile.py]: Hum. Same problem exists in other modules. How about making a helper method or something like that?

Comment (okuta) [inline chainer/function_hooks/cuda_profile.py]: I agree. It is a good suggestion.

Comment (unnonouno) [inline chainer/function_hooks/cuda_profile.py]: I made another PR #2488. Please check it first.

**Comment by unnonouno** [comment]
Merge #2488 before this PR.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**